_posts/2018-03-23-minnowboard-turbot-remote-firmware-flashing-with-remote-testing-environment.md

```diff
@@ -0,0 +1,312 @@
+was `CS#` signal. It was set to `0`, so memory was always enabled, but it was 
+not enough. Minnowboard still doesn't boot. So I decided to use an 
+oscilloscope. That analysis showed me that Minnowboard Turbot doesn't set 
+memory chip's `CS#` input to `0` all the time. It chnges over the time. Mostly
+`CS#` is set to `0`, but periodically it's being set to `1`.
+
+Time analysis of the `CS#` states seemed to be too time-consuming and trying
+correct control of the `CS#` lines too inflexible in the event of changes.
+I needed to reduce the impact of the `CS#` RTE output to make it a Minnowboard
+Turbot took precedence in choosing the state of the `CS#` line.
+I decided to use 1.2 kΩ resistor between RTE `CS#` output and Minnowboard 
+Turbot `CS#` input and this was a good idea.
+
+After resolving `CS#` problem, flashing procedure is very simple.
+It amounts to: 
+1. Turn off Minnowboard platform power supply.
+2. Flash Minnowboard ROM memory via SPI.
+3. Turn on Minnowboard platform power supply.
+
+And it's that all.
+
+### Firmware flashing
+
+I used 2 firmwares to test presented Remote Testing Environment:
+
+[MNW2MAX1.X64.0097.D01.1709211100.bin](https://firmware.intel.com/sites/default/files/MinnowBoard_MAX-0.96-Firmware.Images.zip),
+
+[MNW2MAX1.X64.0097.D01.1709211100.bin](https://firmware.intel.com/sites/default/files/MinnowBoard_MAX-Rel_0_97-Firmware.Images.zip).
+
+
+Code of flashing script:
+```
+#!/bin/bash
+
+# turn off Minnowboard platform power supply
+../rte_ctrl -rel
+
+sleep 1
+
+# flash Minnowboard ROM with a new firmware
+flashrom -p linux_spi:dev=/dev/spidev1.0,spispeed=32000 -w $1
+
+sleep 1
+
+# turn on Minnowboard platform power supply
+../rte_ctrl -rel
+```
+
+> `rte_ctrl` is a program written to control power supply of DUT. For the use 
+shown in this article it can be fully replaced by direct output control of GPIO
+which is being used to turn on and turn off power supply of DUT. In this
+situation flashing script will look like this:
+
+```
+#!/bin/bash
+
+# turn off Minnowboard platform power supply
+echo 0 > /sys/class/gpio/gpio199/value
+
+sleep 1
+
+# flash Minnowboard ROM with a new firmware
+flashrom -p linux_spi:dev=/dev/spidev1.0,spispeed=32000 -w $1
+
+sleep 1
+
+# turn on Minnowboard platform power supply
+echo 1 > /sys/class/gpio/gpio199/value
+```
+
+To verify correctness of firmware flashing executed by RTE I saved
+UEFI Shell logs received before flashing process:
+```
+UEFI Interactive Shell v2.187477C2-69C7-11D2-8E39-00A0C969723B 767BCCA0
+EDK IIlProtocolInterface: 752F3136-4E16-4FDC-A22A-E5F46812F4CA 767BDC58
+UEFI v2.50 (EDK II, 0x00010000)008-7F9B-4F30-87AC-60C9FEF5DA4E 77D637C0
+map: No mapping found.
+Press ESC in 1 seconds to skip startup.nsh or any other key to continue.MnpSyncSendPacket: No network cable detected.
+Shell> 
+```
+
+Then I saved flashing script shown in the earlier part of the article as 
+`flash_mw.sh` in RTE `/root/` directory and executed it.
```

Comment:
@acihy you should name scripts where you create them, relating in such way to previous script can be confusing.